src/clinvar_combiner/snapshot.clj

```diff
@@ -0,0 +1,229 @@
+            [me.raynes.fs :as fs]
+            [taoensso.timbre :as log]
+            [clojure.string :as s]
+            [jackdaw.client :as jc]
+            [clinvar-combiner.config :as config]
+            [clinvar-streams.util :as util])
+  (:import (java.nio.file Paths)
+           (com.google.common.io ByteStreams)
+           (com.google.cloud.storage
+             Bucket BucketInfo Storage StorageOptions
+             BlobId BlobInfo Blob Storage$BlobWriteOption)
+           (java.io FileInputStream)
+           (java.nio.channels FileChannel)
+           (org.apache.kafka.common TopicPartition)
+           (java.time Duration Instant)))
+
+
+(defn download-file
+  "Pull the specified file from cloud storage, writing it locally to target-path"
+  [bucket blob-name target-path]
+  (log/info {:fn :download-file
+             :bucket bucket
+             :blob-name blob-name
+             :target-path target-path})
+  (let [target-path (Paths/get target-path (make-array String 0))
+        blob (.get (.getService (StorageOptions/getDefaultInstance))
+                   (BlobId/of bucket blob-name))]
+    (.downloadTo blob target-path)
+    target-path))
+
+(defn BlobId->str
+  "Formats a BlobId as a fully qualified URI string."
+  [blob-id]
+  (format "gs://%s/%s" (.getBucket blob-id) (.getName blob-id)))
+
+(defn upload-file
+  "Uploads src-file-name to target-bucket as dest-blob-name.
+  Returns the BlobId uploaded to."
+  [target-bucket src-file-name dest-blob-name]
+  (log/info {:fn :upload-file
+             :target-bucket target-bucket
+             :src-file-name src-file-name
+             :dest-blob-name dest-blob-name})
+  (let [gc-storage (.getService (StorageOptions/getDefaultInstance))
+        blob-id (BlobId/of target-bucket dest-blob-name)
+        blob-info (-> blob-id BlobInfo/newBuilder (.setContentType "application/gzip") .build)
+        from (.getChannel (FileInputStream. src-file-name))]
+    (with-open [to (.writer gc-storage blob-info (make-array Storage$BlobWriteOption 0))]
+      (ByteStreams/copy from to))
+    blob-id))
+
+(defn extract-file
+  "Performs tar extract and gzip decompress on the path, writing the output to dest directory."
+  [path dest]
+  (log/info {:fn :extract-file :path path :dest dest})
+  (let [ret (sh "tar" "-xzf" path "-C" dest)]
+    (log/info {:fn :extract-file :msg ret})
+    (if (not= 0 (:exit ret))
+      (throw (ex-info "Failed to extract file" ret))
+      dest)))
+
+(defn archive-file
+  "Performs tar and gzip on the path. Path can be a single file or a directory.
+  Archive will contain the path, so if it is a directory, it will contain the top level
+  directory itself, plus the contents, recursively."
+  [path dest]
+  (log/info {:fn :archive-file :path path :dest dest})
+  (let [ret (sh "tar" "-czf" dest path)]
+    (log/info {:fn :archive-file :msg ret})
+    (if (not= 0 (:exit ret))
+      (throw (ex-info "Failed to archive file" ret))
+      dest)))
+
+(defn trim-suffix
```

Comment:
@theferrit32 nothing critical, but wouldn't use of a regex here be simpler to read and potentially reduce the need for a function? If not, then I'm guessing that Tristan may have a clojure module that holds generic string utility functions like this that could be reused or added to.